RestaurantReservation.Db/Repositories/MenuItemRepository.cs

```diff
@@ -62,14 +62,20 @@ public async Task UpdateAsync(MenuItemModel menuItem)
 
         public async Task DeleteAsync(int id)
         {
-            var menuItem = await _context.MenuItems.FindAsync(id);
+            var menuItem = await _context.MenuItems
+                .Include(m => m.OrderItems)
+                .FirstOrDefaultAsync(m => m.ItemId == id);
+
+            if (menuItem.OrderItems.Any())
+                throw new InvalidOperationException("Cannot delete menu item with existing order items.");
```

Comment:
this is a business rule check that should be moved to the domain/service layer. same comment for the checks in other repos.